ASV-BENCHMARKING.md

```diff
@@ -0,0 +1,82 @@
+Summarised below are the steps required to run the asv benchmark on a new computer.
+
+1. Clone trackintel and checkout to master. Add the trackintel root to python path 
+
+>(**For testing purposes only,** clone the abcnishant007/trackintel and checkout to `asv-trackintel` branch). 
+*@Ye, we need to edit this line after pull request is approved. *
+
+2. Download the larger geolife data from [this branch](https://github.com/abcnishant007/trackintel/tree/benchmark-files) and place it inside `tests/data/geolife_long_10_MB`. This dataset will not be used by default. The dataset option is hardcoded in the benchmark files as shown [here](https://github.com/abcnishant007/trackintel/blob/7b8c2ee2f12d98b59578cd0519aae6a5240ade4c/benchmarks/preprocessing_benchmarks.py#L6). The hardcoding exists temporarily only to ensure that we can switch to the smaller dataset for asv testing purposes on a new computer. Once the whole setup is complete, we can start using the bigger dataset. The hardcoding can be removed (or retained for future setups) and only the `geolife_long_10_MB` data set should be used for actual benchmarks.
+3. Create a new `conda` environment and install the requirements for trackintel. I tested using Ubuntu20 and there were some issues with the requirements. If you are using some other OS, you can skip these are proceed to Step 4
+
+> Some ubuntu specific tweaks:
+
+>1 (a). The `psychopg2` runs into error while doing `pip install -r reqirements.txt`
+
+>> Solution: Use `psychopg2-binary` 
+    
+> 1(b) If you get an error as shown below:
+```
+ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
+spyder 5.1.5 requires pyqt5<5.13, which is not installed.
+spyder 5.1.5 requires pyqtwebengine<5.13, which is not installed.
+
+```
+>> Solution:
+```
+pip install --upgrade --user pyqtwebengine==5.12
+pip install --upgrade --user pyqt5==5.12
+```
+
+4. Install asv using `pip install asv`
+5. Change current directory to the root of trackintel. 
+6. Get the list of commits in short form (last 7 characters of commit id) which were merge commits in master using the shell command below: 
+```git log | grep -B 1 “Merge” | grep “commit” | sed ‘s/commit //g’ | cut -c1-7 | head -n 3 > commits.txt```
```

Comment:
I am actually stuck on this line; my command line (mac) returns `s/commit: invalid command code`. What is this line doing?

Reply:
I changed the `‘s/commit //g’` to `"s/commit //g"` and it works! But the generated `commit.txt` file does not contain any lines

Reply:
On the bash shell, all quotes should be straight, (single or double doesn't matter). Please feel free to try now. I replaced the version with straight ticks and there is a copy option now. Additionally, the whole document has been updated to put the commands in separate code blocks so that they can be copied with a click instead of drag and click copy. That reduces the chances of error. 

I tested for MacOS
 <img width="737" alt="Screenshot 2022-08-01 at 21 41 10" src="https://user-images.githubusercontent.com/9101260/182160943-69f9ada8-90a1-4a86-b8c4-a1be881987f3.png">

The 
```
sed s/`some stuff`/`new stuff`/g 
```
replaced the `<stuff>` with the `<new stuff>`.

Reply:
> I am actually stuck on this line; my command line (mac) returns `s/commit: invalid command code`. What is this line doing?

This line parses the output from the `git log` and extracts the merge commits from them. Finally, cuts them to the last 7 chracters. Why last 7? because asv did not work with full commits; I have no idea why :(